src/baal/active/heuristics/heuristics.py

```diff
@@ -241,27 +272,70 @@ class BatchBALD(BALD):
         Not tested on 4+ dims.
         """
 
-    def __init__(self, num_samples, shuffle_prop=0.0, reduction='none'):
+    def __init__(self, num_samples, num_draw=500, shuffle_prop=0.0, reduction='none'):
         self.epsilon = 1e-5
         self.num_samples = num_samples
+        self.num_draw = num_draw
         super().__init__(shuffle_prop=shuffle_prop,
                          reduction=reduction)
 
+    def _draw_choices(self, probs, n_choices):
+        """
```

Comment:
And a reference to https://github.com/BlackHC/BatchBALD/blob/master/src/torch_utils.py#L187 ?